cmd/kube-scheduler/app/options/options.go

```diff
@@ -0,0 +1,180 @@
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package options
+
+import (
+	"net"
+	"strconv"
+
+	"github.com/golang/glog"
+	"github.com/spf13/pflag"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	apiserveroptions "k8s.io/apiserver/pkg/server/options"
+	utilfeature "k8s.io/apiserver/pkg/util/feature"
+	controlleroptions "k8s.io/kubernetes/cmd/controller-manager/app/options"
+	schedulerappconfig "k8s.io/kubernetes/cmd/kube-scheduler/app/config"
+	"k8s.io/kubernetes/pkg/apis/componentconfig"
+	componentconfigv1alpha1 "k8s.io/kubernetes/pkg/apis/componentconfig/v1alpha1"
+	"k8s.io/kubernetes/pkg/client/leaderelectionconfig"
+)
+
+// Options has all the params needed to run a Scheduler
+type Options struct {
+	// The default values. These are overridden if ConfigFile is set or by values in InsecureServing.
+	ComponentConfig componentconfig.KubeSchedulerConfiguration
+
+	SecureServing           *apiserveroptions.SecureServingOptions
+	CombinedInsecureServing *CombinedInsecureServingOptions
+	Authentication          *apiserveroptions.DelegatingAuthenticationOptions
+	Authorization           *apiserveroptions.DelegatingAuthorizationOptions
+	Deprecated              *DeprecatedOptions
+
+	// ConfigFile is the location of the scheduler server's configuration file.
+	ConfigFile string
+	Master     string
+}
+
+// NewOptions returns default scheduler app options.
+func NewOptions() (*Options, error) {
+	cfg, err := newDefaultComponentConfig()
+	if err != nil {
+		return nil, err
+	}
+
+	hhost, hport, err := splitHostIntPort(cfg.HealthzBindAddress)
+	if err != nil {
+		return nil, err
+	}
+
+	o := &Options{
+		ComponentConfig: *cfg,
+		SecureServing:   nil, // TODO: enable with apiserveroptions.NewSecureServingOptions()
+		CombinedInsecureServing: &CombinedInsecureServingOptions{
+			Healthz: &controlleroptions.InsecureServingOptions{
+				BindNetwork: "tcp",
+			},
+			Metrics: &controlleroptions.InsecureServingOptions{
+				BindNetwork: "tcp",
+			},
+			BindPort:    hport,
+			BindAddress: hhost,
+		},
+		Authentication: nil, // TODO: enable with apiserveroptions.NewDelegatingAuthenticationOptions()
+		Authorization:  nil, // TODO: enable with apiserveroptions.NewDelegatingAuthorizationOptions()
+		Deprecated: &DeprecatedOptions{
+			UseLegacyPolicyConfig:    false,
+			PolicyConfigMapNamespace: metav1.NamespaceSystem,
+		},
+	}
+
+	return o, nil
+}
+
+func splitHostIntPort(s string) (string, int, error) {
+	host, port, err := net.SplitHostPort(s)
+	if err != nil {
+		return "", 0, err
+	}
+	portInt, err := strconv.Atoi(port)
+	if err != nil {
+		return "", 0, err
+	}
+	return host, portInt, err
+}
+
+func newDefaultComponentConfig() (*componentconfig.KubeSchedulerConfiguration, error) {
+	scheme := runtime.NewScheme()
+	componentconfig.AddToScheme(scheme)
+	componentconfigv1alpha1.AddToScheme(scheme)
```

Comment:
in loadConfig above you created a new scheme locally and here are we are doing again?

Reply:
fixed.